### PR TITLE
C4-45 Slight MPIndexer refactor

### PR DESCRIPTION
**Pull request by willronchetti**
- See C4-45
- Small cleanup in the MPIndexer so that we can recover from exceptions within the multiprocessing library, which apparently we trigger somehow

### CODE REVIEW

File: snovault/elasticsearch/mpindexer.py [hunk 1]
```diff
@@ -232,24 +232,26 @@ def update_objects(self, request, counter):
             # add more jobs if any are finished and indexing is ongoing
             while True:
                 results_to_add = []
-                idxs_to_rm = []
                 for idx, res in enumerate(async_results):
-                    if res.ready():
```
Comment (netsettler) [inline snovault/elasticsearch/mpindexer.py]: Are you sure you don't need to still test this? If it was only previously getting all this info when something was ready, won't it now try to get un-ready values? Will that raise an error?

Comment (willronchetti) [reply snovault/elasticsearch/mpindexer.py]: Good catch. `except TimeoutError` is required here for this function correctly. Will fix.

Comment (netsettler) [reply snovault/elasticsearch/mpindexer.py]: Looks like the subsequent changes resolved this.

File: snovault/util.py [hunk 2]
```diff
@@ -37,7 +37,9 @@ def dictionary_lookup(dictionary, key):
     dictionary_lookup(d, k) is the same as d[k] but with more informative error reporting.
     """
     if not isinstance(dictionary, dict) or (key not in dictionary):
-        raise DictionaryKeyError(dictionary=dictionary, key=key)
+        log.error('Got dictionary KeyError with %s and %s' % (dictionary, key))
+        return None
```
Comment (netsettler) [inline snovault/util.py]: Yeah, we should come up with a way to then notice that these logged things are happening. But it's fine. I've seen this sort of problem before and can imagine why it's a problem. It probably happens in a place where the error is hard to catch.

Comment (willronchetti) [reply snovault/util.py]: This is a really strange problem IMO. Perhaps we should discuss this further and come up with a better solution. FYI this log statement will show up in the application error logs on the server. You could grep for `util` to find them (if they exist).

**Comment by netsettler** [comment]
This needs a version bump in `pyproject.toml`. It's new functionality, so I'd bump the minor version.

**Pull request merged**